src/OrchardCore.Modules/OrchardCore.Admin/AdminZoneFilter.cs

```diff
@@ -1,29 +1,34 @@
-using System;
-using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
 using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System;
+using System.Threading.Tasks;
 
 namespace OrchardCore.Admin
 {
     /// <summary>
-    /// This filter makes an controller that starts with Admin behave as
+    /// This filter makes an controller that starts with Admin and Razor Pages in /Pages/Admin folder behave as
     /// it had the <see cref="AdminAttribute"/>.
     /// </summary>
-    public class AdminZoneFilter : IActionFilter
+    public class AdminZoneFilter : IAsyncResourceFilter
     {
-        public void OnActionExecuted(ActionExecutedContext context)
-        {
-        }
-
-        public void OnActionExecuting(ActionExecutingContext context)
+        public Task OnResourceExecutionAsync(ResourceExecutingContext context, ResourceExecutionDelegate next)
         {
-            var controller = context.Controller as Controller;
-            if (controller != null)
+            if(context.ActionDescriptor is ControllerActionDescriptor)
             {
-                if(controller.GetType().Name.StartsWith("Admin", StringComparison.OrdinalIgnoreCase))
+                if( ((ControllerActionDescriptor)context.ActionDescriptor).ControllerName.StartsWith("Admin", StringComparison.OrdinalIgnoreCase) )
                 {
                     AdminAttribute.Apply(context.HttpContext);
                 }
             }
+            else if(context.ActionDescriptor is PageActionDescriptor)
+            {
+                if( ((PageActionDescriptor)context.ActionDescriptor).ViewEnginePath.StartsWith("/Admin", StringComparison.OrdinalIgnoreCase) )
+                {
```

Comment:
So here to see if it is an admin page action, you check that it is under the 1st level admin folder, this in place of checking the razor page file name (without extensions) e.g `AdminHello`, i'm fine with this, @deanmarcussen ?

Then i think that here also we can be case insensitive, @deanmarcussen ?

Here in my PR i ended up to use `descriptor.DisplayText` that was always with the right letter cases e.g `/Admin/AdminHello`, i don't remember with `ViewEnginePath` but if it's the case that's okay.

So finally you could just use

    if (pageDescriptor.ViewEnginePath.StartsWith("/Admin"))

Reply:
yes I agree we can go case insensitive here, and also that we go with the admin folder name convention, rather than checking the razor page file name. seems to make sense to me.

Reply:
I guess having Admin folder ( `/Admin/` ) at any level is good enough

Reply:
@deanmarcussen @ns8482e okay cool

@ns8482e then just use `if (pageDescriptor.ViewEnginePath.Contains("/Admin/"))`

Reply:
@jtkech  @deanmarcussen Here, For controller, I wasn't sure if you want me to change following 

`action.ControllerName.StartsWith("Admin", StringComparison.OrdinalIgnoreCase)`

To, 

`action.ControllerName == "Admin"`

Reply:
`action.ControllerName == "Admin"`